debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlTaskContext.java

```diff
@@ -235,8 +235,9 @@ protected SnapshotMode snapshotMode() {
         return SnapshotMode.parse(value, MySqlConnectorConfig.SNAPSHOT_MODE.defaultValueAsString());
     }
 
-    public boolean useMinimalSnapshotLocking() {
-        return config.getBoolean(MySqlConnectorConfig.SNAPSHOT_MINIMAL_LOCKING);
+    public MySqlConnectorConfig.SnapshotLockingMode snapShotLockingMode() {
```

Comment:
Please expose this as a getter on `MySqlConnectorConfig` instead (following the pattern from `CommonConnectorConfig`). This is an ongoing refactoring we're pursuing, all the options should be exposed via getters on the connector's sub-class of `CommonConnectorConfig`.

Reply:
I feel like I'm missing something here.  Should this method go completely away?  Or should this method remain and instead just call a getter on MySqlConnectorConfig?

If this method goes completely away, how do I access the getter from within SnapshotReader?  Would I need to cast Configuration after calling taskContext.config() from SnapshotReader's constructor?

Reply:
Yes, it should go away completely. Instead, there should be a getter for that property on `MySqlConnectorConfig` and that config itself should be made available via `MySqlTaskContext#getConnectorConfig()`. I thought the latter would already be the case, but instead it's on `MySqlJdbcContext`. I think it can be removed there and should just be living on `MySqlTaskContext`.

Reply:
Thanks for the clarification!

Reply:
My pleasure; we're in the process of unifying some more bits and pieces across the different connectors. So things are a bit in flux right now and can be a bit confusing. But I hope it'll be better eventually :)